src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/CLinker.java

```diff
@@ -26,23 +26,65 @@
 package jdk.incubator.foreign;
 
 import jdk.internal.foreign.NativeMemorySegmentImpl;
+import jdk.internal.foreign.PlatformLayouts;
 import jdk.internal.foreign.Utils;
 import jdk.internal.foreign.abi.SharedUtils;
 
-import java.nio.ByteOrder;
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodType;
 import java.nio.charset.Charset;
 import java.util.Objects;
 import java.util.function.Consumer;
 
+import static jdk.internal.foreign.PlatformLayouts.*;
+
 /**
- * A set of utilities for working with libraries using the C language/ABI
+ * This interface is a used as a super class for all ForeinLinker implementations that
+ * implement the system's C ABI and calling convention. As such, instances of this interface
+ * can be used to link foreign functions in native libraries that follow the JVM's target
+ * platform C application binary interface (ABI).
+ *
+ * <p>There are two components that go into linking a foreign function: a method type, and
+ * a function descriptor. The method type, consisting of a set of 'carrier' types constitutes
+ * the Java side of a call to a foreign function, while the function descriptor constitutes
+ * the native side of the call. Memory layout attributes are used in the function descriptor
+ * to attach ABI classification meta-data to memory layouts, which are required for linking.
+ * Clients of this API should use the prepared memory layout constants found in this interface
```

Comment:
Clients of this API should build function descriptors using the predefined memory layout constants (based on a subset of the built-in types provided by the C language), found in this interface; a failure to do so might result in linkage errors, given that this linker requires additional classification information to determine e.g. how arguments should be shuffled during a foreign function call.